nussl/core/audio_signal.py

```diff
@@ -1644,6 +1649,311 @@ def embed_audio(self, ext='.mp3', display=True):
         from . import play_utils
         return play_utils.embed_audio(self, ext=ext, display=display)
 
+    ##################################
+    #          Effect Hooks          #
+    ##################################
+
+    def build_effects(self, reset=True, overwrite=False):
+        """
+        Builds all effects in the AudioSignal's effects chains using the audio data of the calling 
+        AudioSignal. If reset=True, the current effects chains will be deleted. If overwrite=True, 
```

Comment:
What exactly does `reset=True` do? Does it clear the effects chain before the effects are applied if there are any left over from last time?

Reply:
Removes effects in effects chains after applying effects. Made this more clear in docs